src/native/managed/cdacreader/tests/MockMemorySpace.cs

```diff
@@ -106,24 +114,96 @@ public Builder AddHeapFragments(IEnumerable<HeapFragment> fragments)
             return this;
         }
 
-        public ReadContext Create()
+        private HeapFragment CreateContractDescriptor(int jsonLength, int pointerDataCount)
         {
-            if (_created)
-                throw new InvalidOperationException("Context already created");
-            GCHandle fragmentReaderHandle = default; ;
-            if (_heapFragments.Count > 0)
+            Span<byte> descriptor = stackalloc byte[_targetTestHelpers.ContractDescriptorSize];
+            _targetTestHelpers.ContractDescriptorFill(descriptor, jsonLength, pointerDataCount);
+            const ulong ContractDescriptorAddr = 0xaaaaaaaa;
```

Comment:
Is this an important value? The address is also only 32-bits and we are assigning to a 64-bit value. Perhaps a comment to clarify what this is used for?

Reply:
Moved it to the top of the file and added a comment that it's arbitrary but mustn't overlap any of the other arbitrary addresses that the tests use.

Right now in the interest of not doing too much magic and keeping the tests reproducible all the "allocation" in the mock memory space is done by hand - by picking an address that doesn't overlap anything and allocating a fragment there (the builder will tell you if you mathed wrong).  I think in the longer term it will be easier to write bigger tests if we had some kind of mock malloc, but I'm worried that debugging a failed test will be too difficult, or the tests might become non-deterministic or something.  But on the other hand, I'm hoping after the contract refactoring we won't _have_ to write massive tests.  So we'll see.